### PR TITLE
Add inspect_all for dandisets section (auto-path inference)

### LINKED ISSUE #190: [Documentation]: demonstrate inspect_all for dandiset
### What would you like changed or added to the documentation and why?

demonstrate inspect_all for dandiset

### Do you have any interest in helping write or edit the documentation?

No.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this change was not already [requested](https://github.com/neurodatawithoutborders/nwbinspector/issues)?

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fixes #190

### CODE REVIEW

File: docs/user_guide/using_the_library.rst [hunk 1]
```diff
@@ -55,6 +55,41 @@ This has the same return structure as :py:class:`~nwbinspector.nwbinspector.insp
 
 
 
+.. _simple_streaming_api:
+
+Inspect a DANDI set (ROS3)
+--------------------------
+
+It is a common use case to want to inspect and review entire datasets of NWB files that have already been
+uploaded to the :dandi-archive:`DANDI Archive <>`. While it is possible to simply download the entire DANDI set to your local computer and
```
Comment (bendichter) [inline docs/user_guide/using_the_library.rst]:
```suggestion
uploaded to the :dandi-archive:`DANDI Archive <>`. While it is possible to simply download the entire dandiset to your local computer and
```

File: docs/user_guide/using_the_library.rst [hunk 2]
```diff
@@ -55,6 +55,41 @@ This has the same return structure as :py:class:`~nwbinspector.nwbinspector.insp
...
+
+.. code-block:: python
+
+    from nwbinspector import inspect_all
+
+    dandiset_id = "..."  # for example, 000004
+
+    messages = list(inspect_all(nwbfile_path=dandiset_id, stream=True))
+
+If there are multiple versions of the DANDI set available (*e.g.*, separate 'draft' and 'published' versions) you can additionally specify this with the ``version_id`` argument...
```
Comment (bendichter) [inline docs/user_guide/using_the_library.rst]:
```suggestion
If there are multiple versions of the dandiset available (*e.g.*, separate 'draft' and 'published' versions) you can additionally specify this with the ``version_id`` argument...
```

File: docs/user_guide/using_the_library_advanced.rst [hunk 3]
```diff
@@ -45,18 +45,13 @@ Of course, the generator can be treated like any other iterable as well, such as
...
-TB scale!
+.. _advanced_streaming_api:
 
-The general tutorial for using the :code:`ros3` driver can be found :ros3-tutorial:`here <>`. This driver can be passed
-directly into our core inspection functions, and the ``path`` or ``nwbfile_path`` arguments in this case become the
-S3 path on the DANDI archive. Resolution of these paths can be performed via the following code
+Fetching and inspecting individual DANDI assets (ROS3)
+------------------------------------------------------
+
+While the section explaining :ref:`basic steaming of a DANDI set<simple_streaming_api>` covered the simplest and most convenient usage of the streaming feature, sometimes a greater degree of control or customization is required. The :py:attr:`driver` argument of the :py:class:`~pynwb.NWBHDF5IO` can be passed directly into our core inspection functions. In this case, the :py:attr:`path` or :py:attr:`nwbfile_path` arguments become the full S3 path on the DANDI archive. Resolution of these paths can be performed via the following code...
```
Comment (bendichter) [inline docs/user_guide/using_the_library_advanced.rst]:
```suggestion
While the section explaining :ref:`basic steaming of a dandiset <simple_streaming_api>` covered the simplest and most convenient usage of the streaming feature, sometimes a greater degree of control or customization is required. The :py:attr:`driver` argument of the :py:class:`~pynwb.NWBHDF5IO` can be passed directly into our core inspection functions. In this case, the :py:attr:`path` or :py:attr:`nwbfile_path` arguments become the full S3 path on the DANDI archive. Resolution of these paths can be performed via the following code...
```

File: docs/user_guide/using_the_library.rst [hunk 4]
```diff
@@ -55,6 +55,41 @@ This has the same return structure as :py:class:`~nwbinspector.nwbinspector.insp
...
+
+Inspect a DANDI set (ROS3)
+--------------------------
+
+It is a common use case to want to inspect and review entire datasets of NWB files that have already been
+uploaded to the :dandi-archive:`DANDI Archive <>`. While it is possible to simply download the entire dandiset to your local computer and
+run the NWB Inspector as usual, there is an alternative which is less expensive in terms of bandwith. This can be especially
+useful when the DANDI set is large and impractical to download in full, such as one that is multiple terabytes in total size.
+
+A general tutorial for using the :code:`ros3` driver can be found in the :ros3-tutorial:`PyNWB documentation <>` - however, the NWB Inspector has automated functionality for resolving the technical identification of assets such that the only piece of information needed is the six-digit DANDI set identifier...
```
Comment (bendichter) [inline docs/user_guide/using_the_library.rst]: careful with the word "however". I don't think it's appropriate in this case.

Comment (CodyCBakerPhD) [reply docs/user_guide/using_the_library.rst]: Fixed

**Pull request merged**